static/app/components/editableText.tsx

```diff
@@ -132,121 +131,59 @@ function EditableText({
-              />
-            </StyledField>
-            <InputLabel>{inputValue}</InputLabel>
-          </InputWrapper>
-        ) : (
-          <React.Fragment>
-            <Label
-              onClick={isDisabled ? undefined : handleEditClick}
-              ref={labelRef}
-              data-test-id="editable-text-label"
-            >
-              <InnerLabel>{inputValue}</InnerLabel>
-            </Label>
-            {!isDisabled && <StyledIconEdit />}
-          </React.Fragment>
-        )}
-      </InnerWrapper>
+    <Wrapper ref={innerWrapperRef} isDisabled={isDisabled} isEditing={isEditing}>
+      {isEditing ? (
+        <InputWrapper isEmpty={isEmpty} data-test-id="editable-text-input">
+          <StyledInput
+            name={name}
+            ref={inputRef}
+            value={inputValue}
+            onChange={handleInputChange}
+          />
+          <InputLabel>{inputValue}</InputLabel>
+        </InputWrapper>
+      ) : (
+        <React.Fragment>
+          <Label
+            onClick={isDisabled ? undefined : handleEditClick}
+            ref={labelRef}
+            data-test-id="editable-text-label"
+          >
+            <InnerLabel>{inputValue}</InnerLabel>
+            {!isDisabled && <IconEdit />}
+          </Label>
+        </React.Fragment>
+      )}
     </Wrapper>
   );
 }
 
 export default EditableText;
 
 const Label = styled('div')`
-  display: inline-block;
-  border-radius: ${p => p.theme.borderRadius};
-  text-align: left;
-  padding-left: 10px;
-  height: 40px;
-  max-width: 100%;
+  display: grid;
+  grid-auto-flow: column;
+  align-items: center;
+  gap: ${space(1)};
 `;
 
 const InnerLabel = styled(TextOverflow)`
   border-top: 1px solid transparent;
   border-bottom: 1px dotted ${p => p.theme.gray200};
-  transition: border 150ms;
-  height: 40px;
-  line-height: 38px;
-`;
-
-const StyledIconEdit = styled(IconEdit)`
-  height: 40px;
-  position: absolute;
-  right: 0;
-`;
-
-const Wrapper = styled('div')`
-  display: flex;
-  justify-content: flex-start;
-  height: 40px;
-`;
-
-const InnerWrapper = styled('div')<{isDisabled: boolean; isEditing: boolean}>`
-  position: relative;
-  display: inline-flex;
-  max-width: 100%;
-
-  ${p =>
-    p.isDisabled
-      ? `
-          ${StyledIconEdit} {
-            cursor: default;
-          }
-
-          ${InnerLabel} {
-            border-bottom-color: transparent;
-          }
-        `
-      : `
-       ${!p.isEditing && `padding-right: 25px;`}
-        :hover {
-          padding-right: 0;
-          ${StyledIconEdit} {
-            display: none;
-          }
-          ${Label} {
-            background: ${p.theme.gray100};
-            padding: 0 14px 0 10px;
-          }
-          ${InnerLabel} {
-            border-bottom-color: transparent;
-          }
-        }
-      `}
 `;
 
 const InputWrapper = styled('div')<{isEmpty: boolean}>`
-  position: relative;
-  min-width: ${p => (p.isEmpty ? '100px' : '50px')};
```

Comment:
I wasn't sure about it either, and I wanted to review it again, but shouldn't the empty field be less wide?

Reply:
Before:

![image](https://user-images.githubusercontent.com/29228205/116443389-e0ab3280-a853-11eb-8c58-81a8ca0cd05c.png)

After:
![image](https://user-images.githubusercontent.com/29228205/116443437-f15ba880-a853-11eb-81b7-c8cfb69067d5.png)

Reply:
There's a min-width that's already set on the input which is 265px. TBH, I'm not sure where that width is coming from. It might even be a bootstrap style.